matrix-sdk/src/main/java/org/matrix/androidsdk/crypto/algorithms/olm/MXOlmDecryption.java

```diff
@@ -89,6 +93,82 @@ public MXDecryptionResult decryptEvent(Event event) {
             result = new MXDecryptionResult();
             result.mPayload = new JsonParser().parse(JsonUtils.convertFromUTF8(payloadString));
 
+            if (null != result.mPayload) {
+                JsonObject payloadAsJSon = result.mPayload.getAsJsonObject();
+
+                if (!payloadAsJSon.has("recipient")) {
+                    Log.e(LOG_TAG,  "Olm event (id=" + event.eventId + ") contains no " +"'recipient' property; cannot prevent unknown-key attack");
+                } else {
+                    try {
+                        String recipient = payloadAsJSon.get("recipient").getAsString();
+
+                        if (!TextUtils.equals(recipient, mSession.getMyUserId())) {
+                            Log.e(LOG_TAG, "Event " + event.eventId + ": Intended recipient " + recipient + " does not match our id " + mSession.getMyUserId());
+                            result.mPayload = null;
+                            result.mCryptoError = new MXCryptoError(MXCryptoError.MESSAGE_NOT_INTENDED_FOR_THIS_DEVICE);
+                            return result;
+                        }
+                    } catch (Exception e) {
+                        Log.e(LOG_TAG, "## decryptEvent() : " + e.getMessage());
+                    }
+                }
+
+                if (!payloadAsJSon.has("recipient_keys")) {
```

Comment:
This PR checks recipient_keys on reception but never fills the value on sending. Is it normal?